### ISSUE TITLE
Warning/notice when creating a post to a defederated instance

### ISSUE THREAD

**Issue by 0WN463** [LemmyNet/lemmy-ui#2680]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

(Still new to Lemmy, so pardon me if any of my terminologies are not accurate)

When creating a post to a community on a defederated instance, there is no warning for the user that the post will not be shown to the instance.

I was confused previously about several of my post not appearing on the target instance/community after posting them, which now I learnt is due to defederation.

I feel we should add a disclaimer or some notice to the user when trying to post to a community on defederated instance so that they are aware of this behaviour, especially since the instance dropdown in the post creation screen still contains communities from defederated instances.

**Comment by dessalines** [comment]
If your server is the one that's blocking another one, I believe we already throw a back-end error there.

But if its the receiving server, then its probably not doable. Your server doesn't know which servers are currently blocking it, since its blocked on the receiving side. So I don't know any way we could do this.

**Comment by dessalines** [comment]
> From https://lemmy.cafe/instances, we see that lemmy.cafe blocked lemmy.ml, but I am still able to create a post from lemmy.cafe, but I was extremely confused cause my post didn't appear under lemmy.ml's search, but it appears in lemmy.cafe.

That is a case we could potentially handle, since your instance knows about the block. We can't do the reverse direction though.

**Comment by dessalines** [comment]
Inactive doesn't necessarily mean de-fed. We don't know which servers have blocked ours, as we don't go out crawling for that info.

**Comment by Die4Ever** [comment]
> Inactive doesn't necessarily mean de-fed. We don't know which servers have blocked ours, as we don't go out crawling for that info.

I agree inactive doesn't mean defed, but a general warning for trying to post to an unreachable server makes sense to me. The end result is the same, it won't work lol, generally it's a good idea to have a warning for something that is unlikely to work correctly.

**Comment by dessalines** [comment]
This isn't something we can check in the UI.
